tests/fixtures/custom/custom-route-generator/serverlessRouteGenerator.ts

```diff
@@ -0,0 +1,121 @@
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
+import * as fs from 'fs';
+import * as handlebars from 'handlebars';
+import { ExtendedRoutesConfig } from '@tsoa/cli/src/cli';
+import { Tsoa, TsoaRoute, assertNever } from '@tsoa/runtime';
+import { fsReadFile, fsWriteFile, fsExists, fsMkDir } from '@tsoa/cli/src/utils/fs';
+import { AbstractRouteGenerator } from '@tsoa/cli/src/routeGeneration/routeGenerator';
+import path = require('path');
+
+export interface ServerlessRoutesConfig extends ExtendedRoutesConfig {
+  modelsTemplate?: string;
+  modelsFileName?: string;
+  handlerTemplate?: string;
+}
+
+export default class ServerlessRouteGenerator extends AbstractRouteGenerator<ServerlessRoutesConfig> {
+  constructor(metadata: Tsoa.Metadata, options: ServerlessRoutesConfig) {
+    super(metadata, options);
+    this.registerTemplateHelpers();
+  }
+
+  protected pathTransformer(path: string): string {
+    return path;
+  }
+
+  registerTemplateHelpers() {
+    handlebars.registerHelper('json', (context: any) => {
+      return JSON.stringify(context);
+    });
+
+    const additionalPropsHelper = (additionalProperties: TsoaRoute.RefObjectModelSchema['additionalProperties']) => {
+      if (additionalProperties) {
+        // Then the model for this type explicitly allows additional properties and thus we should assign that
+        return JSON.stringify(additionalProperties);
+      } else if (this.options.noImplicitAdditionalProperties === 'silently-remove-extras') {
+        return JSON.stringify(false);
+      } else if (this.options.noImplicitAdditionalProperties === 'throw-on-extras') {
+        return JSON.stringify(false);
+      } else if (this.options.noImplicitAdditionalProperties === 'ignore') {
+        return JSON.stringify(true);
+      } else {
+        return assertNever(this.options.noImplicitAdditionalProperties);
+      }
+    };
+    handlebars.registerHelper('additionalPropsHelper', additionalPropsHelper);
+  }
+
+  public async GenerateCustomRoutes() {
+    if (!fs.lstatSync(this.options.routesDir).isDirectory()) {
+      throw new Error(`routesDir should be a directory`);
+    }
+    console.log('Generating models');
+    await this.generateModels();
+    await this.generateRoutes();
+    await this.generateStack();
+  }
+
+  async shouldWriteFile(fileName: string, content: string) {
```

Comment:
I think extracting this also makes sense in the default (as a private method, not for overriding)